frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -279,7 +279,8 @@ export const insightLogic = kea<insightLogicType>({
                                 cache.abortController.signal
                             )
                         } else if (insight === InsightType.FUNNELS) {
-                            response = await pollFunnel(currentTeamId, params)
+                            const { new_entity, ...restParams } = params
```

Comment:
I don't think this will solve https://sentry.io/organizations/posthog2/issues/3351353463/?referrer=slack , because the `step.order` will still be out of whack?

I think for the frontend fix, it makes sense to not send requests at all, when new_entity is not empty?

The backend is at fault too, tbh, using the deprecated `step.order` instead of `step.index` which would resolve this with your current change. I'll put up a backend PR too; but think it's still worth fixing the frontend to not send requests with new_entities non-empty.

Reply:
The reason it's sending it is because a user adds multiple new entity options and then changes one of them to a real filter. So a request should still be sent

Reply:
I'm confused, what do you mean by not sending requests with new_entities non-empty? Like, don't send requests if there's a new_entities param?

Reply:
That works.

Although, I don't think it's a question of consistency, but how the different insights behave:

A funnel is never complete if there's one or more "select filters" involved.
A trend graph is complete as long as there's one non-empty selector involved.

---

At the same time, I'm all for doing it for trends as well, if it reduces confusion :) I worry this might add to the confusion though, as with trends I can have this question of: "I choose two lines, haven't chosen a third yet, but why can't I see the two existing lines?", while it's not the same for a funnel.

Reply:
i'll just add it to funnels

Reply:
nvm there's annoying edge cases to consider if we do that too like deleting the empty filters won't cause the query to re-run

Reply:
I'll have a look when I get the chance 👀 . This still remains borked though :$